src/cr/cube/matrix.py

```diff
@@ -97,14 +93,44 @@ def _assembled_rows(self):
 
         Each row vector also reflects any new elements introduced by inserted columns.
         """
-        opposing_insertions = self._inserted_columns
+        return self._assembled_vectors(
+            self._base_rows, self._inserted_rows, self._inserted_columns
+        )
+
+    def _assembled_vectors(
```

Comment:
This method seems to be generic for row or columns, but why the iteration in the return statement is based on `cols`?

Reply:
Yes, good catch, that should be vector, I've made the fix, thanks! :)